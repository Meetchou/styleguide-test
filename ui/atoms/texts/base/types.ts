export type HProps = {
  className?: string,
  children: React.ReactNode;
};
