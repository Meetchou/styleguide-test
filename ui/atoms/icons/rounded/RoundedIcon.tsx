import React, { FC } from 'react';
import cls from 'classnames';

import Icon from '@ui/atoms/icons/icon/Icon';
import { IconType, ColorType, KeyOfColorType } from '@ui/types';

export type RoundedIconProps = {
  className?: string;
  color: ColorType;
  icon: IconType;
};

export const roundedIconTheme: KeyOfColorType = {
  success: 'bg-success-100 text-success-500',
  danger: 'bg-danger-100 text-danger-500',
  primary: 'bg-primary-100 text-primary-500'
};

const RoundedIcon:FC<RoundedIconProps> = ({
  className,
  icon,
  color
}) => (
  <div className={cls(
    'rounded-full p-1 fill-current',
    roundedIconTheme?.[color] || '',
    className
  )}>
    <Icon icon={icon} className='w-4 h-4' />
  </div>
);

export default RoundedIcon;
