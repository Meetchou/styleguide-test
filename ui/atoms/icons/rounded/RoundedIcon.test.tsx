import React from 'react';

import render from '@ui/testing/render';

import { ColorType, IconType } from '@ui/types';

import RoundedIcon from './RoundedIcon';

describe('ui/atoms/icons/rounded/RoundedIcon', () => {
  const props: { icon: IconType, color: ColorType } = {
    icon: 'check',
    color: 'primary'
  };

  it('renders correctly', () => {
    const tree = render(<RoundedIcon {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
