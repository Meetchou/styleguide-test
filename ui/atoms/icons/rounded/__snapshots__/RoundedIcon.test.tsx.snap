// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/atoms/icons/rounded/RoundedIcon renders correctly 1`] = `
<div
  className="rounded-full p-1 fill-current bg-primary-100 text-primary-500"
>
  <svg
    className="w-4 h-4"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
    />
  </svg>
</div>
`;
